contracts/contracts/token/OUSD.sol

```diff
@@ -511,64 +507,78 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
      * address's balance will be part of rebases and the account will be exposed
      * to upside and downside.
      */
-    function rebaseOptIn() public nonReentrant {
+    function rebaseOptIn() external {
         _rebaseOptIn(msg.sender);
     }
 
     function _rebaseOptIn(address _account) internal {
-        require(_isNonRebasingAccount(_account), "Account has not opted out");
-
-        // Convert balance into the same amount at the current exchange rate
-        uint256 newCreditBalance = _creditBalances[_account]
-            .mul(_rebasingCreditsPerToken)
-            .div(_creditsPerToken(_account));
-
-        // Decreasing non rebasing supply
-        nonRebasingSupply = nonRebasingSupply.sub(balanceOf(_account));
+        // prettier-ignore
+        require(
+            alternativeCreditsPerToken[_account] > 0 || (
+                _account.code.length > 0 && // isContract
```

Comment:
From a requirements point of view on this function, there's no reason that a contract is any different here. We can allow contracts and non-contracts alike to rebaseOptIn. We should be able to remove this one line. We can also just keep the require text down to "Account must be non-rebasing"

Reply:
https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/24865d02a9f0e29116f72b71490eed0dbee375df